aiida/backends/tests/cmdline/commands/test_graph.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+import errno
+import os
+
+import click
+from click.testing import CliRunner
+
+from aiida.backends.testbase import AiidaTestCase
+from aiida.cmdline.commands import graph
+
+def delete_temporary_file(filepath):
+    """
+    Attempt to delete a file, given an absolute path. If the deletion fails because the file does not exist
+    the exception will be caught and passed. Any other exceptions will raise.
+
+    :param filepath: the absolute file path
+    """
+    try:
+        os.remove(filepath)
+    except OSError as exception:
+        if exception.errno != errno.ENOENT:
+            raise
+        else:
+            pass
+
+class TestVerdiGraph(AiidaTestCase):
+    
+    @classmethod
+    def setUpClass(cls):
+        super(TestVerdiGraph, cls).setUpClass()
+        from aiida.orm import Node, LinkType 
+       
+        cls.node = Node().store()
+
+    def setUp(self):
+        self.cli_runner = CliRunner()
+
+    def test_generate_graph(self):
+        """
+        Test that the default graph can be generated
+        The command should run without error and should produce the .dot file
+        """
+        # Get a PK of a node which exists
+        root_node = str(self.node.pk)
+        filename = root_node + '.dot'
+        options = [root_node]
+        try:
+            result = self.cli_runner.invoke(graph.generate, options)
+            self.assertIsNone(result.exception)
+            self.assertTrue(os.path.isfile(filename))
+        finally:
+            delete_temporary_file(filename)
+
+    def test_catch_bad_pk(self):
+        """
+        Test that an invalid root_node pk (non-numeric, negative, or decimal),
+        or non-existant pk will produce an error        
+        """
+        from aiida.orm import load_node 
+        
+        # Forbidden pk
+        for root_node in ['xyz','-5','3.14'] :
```

Comment:
Formatting: there should be a space after each comma and no space between the list and the colon. I think because the regex in the `.pre-commit.yml` is not correct, your pre-commit formatter did not actually run. Or you did not add this file yet

Reply:
It hadn't been staged so was not formatted. Done now